base/iterator.jl

```diff
@@ -559,3 +559,52 @@ function next(itr::PartitionIterator, state)
     end
     return resize!(v, i), state
 end
+
+
+immutable Concat{I,T,S}
+    first::T
+    itr::I
+    st::S
+end
+
+function Concat(c)
+    s = start(c)
+    done(c, s) && error("argument to Concat(it) must contain at least one element")
```

Comment:
`throw(ArgumentError("argument to Concat must contain at least one element"))` would be more consistent with other iterator constructor error conditions